components/icu/examples/date_try_from_fields.rs

```diff
@@ -0,0 +1,41 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+#![no_main]
+use icu::calendar::Ref;
+icu_benchmark_macros::instrument!();
+use icu_benchmark_macros::println;
+
+use icu_calendar::types::DateFields;
+use icu_calendar::{AnyCalendar, AnyCalendarKind, Date};
+
+const CALENDAR_KINDS: &[AnyCalendarKind] = &[
+    AnyCalendarKind::Buddhist,
+    AnyCalendarKind::Chinese,
+    AnyCalendarKind::Gregorian,
+    AnyCalendarKind::Indian,
+    AnyCalendarKind::Japanese,
+    AnyCalendarKind::Ethiopian,
+];
+
+fn main() {
+    for &kind in CALENDAR_KINDS {
+        let cal = AnyCalendar::new(kind);
+
+        let mut fields = DateFields::default();
+        fields.extended_year = Some(2025);
+        fields.month_code = Some(b"M07");
+        fields.day = Some(8);
+
+        let date = Date::try_from_fields(fields, Default::default(), Ref(&cal));
+
+        let kind_str = format!("{:?}", kind);
+        let date_str = match date {
+            Ok(date) => format!("{:?}", date),
+            Err(err) => format!("Error: {:?}", err),
+        };
+
+        println!("Constructed date for {:?} = {:?}", kind_str, date_str);
```

Comment:
using `format!` defeats the purpose of `icu_benchmark_macros` and will pull in all of `std::fmt`. Use the `println!` macro with a type that implements `Writeable`, such as `u8`.

```suggestion
        println!("The day is {}", date.unwrap().day_of_month().0);
```

Reply:
Please let's not use `.unwrap()` though. How about

```rust
match date {
    Ok(date) => println!("The day is {}", date.day_of_month().0),
    Err(_e) => println!("Not a valid date")
}
```